client.go

```diff
@@ -0,0 +1,107 @@
+package growthbook
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"net/http"
+)
+
+const defaultApiHost = "https://cdn.growthbook.io"
+
+var (
+	ErrNoDecryptionKey = errors.New("No decryption key provided")
+)
+
+type Client struct {
+	data             *data
+	enabled          bool
+	attributes       Attributes
+	url              string
+	forcedVariations ForcedVariationsMap
+	qaMode           bool
+	trackingCallback (TrackingCallback)
+	logger           *slog.Logger
+}
+
+// ForcedVariationsMap is a map that forces an Experiment to always assign a specific variation. Useful for QA.
+type ForcedVariationsMap map[string]int
+
+// TrackingCallback function that is executed every time a user is included in an Experiment.
+type TrackingCallback func(*Experiment, *Result)
+
+func NewApiClient(apiHost string, clientKey string) (*Client, error) {
+	ctx := context.Background()
+	return NewClient(ctx, WithApiHost(apiHost), WithClientKey(clientKey))
+}
+
+func NewClient(ctx context.Context, opts ...ClientOption) (*Client, error) {
+	client := defaultClient()
+	for _, opt := range opts {
+		err := opt(client)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return client, nil
+}
+
+func defaultClient() *Client {
+	return &Client{
+		data: &data{
+			apiHost:    defaultApiHost,
+			httpClient: http.DefaultClient,
+		},
+		enabled: true,
+		qaMode:  false,
+		logger:  slog.Default(),
+	}
+}
+
+// SetFeatures updates shared client features.
+func (client *Client) SetFeatures(features FeatureMap) error {
+	client.data.mu.Lock()
+	defer client.data.mu.Unlock()
+
+	client.data.features = features
+	return nil
+}
+
+// SetJSONFeatures updates shared features from JSON
+func (client *Client) SetJSONFeatures(featuresJSON string) error {
```

Comment:
Normally we will have background process (data provider) that will update client shared data. It will be available as another `ClientOption` like `WithSSEDataProvider()` or `WithPollingDataProvider(pollingtime)`. They all need api host, client key, decryption key and http client, so I decided to add it as common options instead of parameters of option generators. 

Regarding `SetXXXFeatures` public methods. I just thought it could be useful for testing purposes, or in case when you need some custom login in data provider, so you can just get features JSON from somewhere and put it into client state.